### ISSUE TITLE
increase code coverage, especially unit tests

### ISSUE THREAD

**Issue by anilbey** [BlueBrain/EModelRunner#57]
The coverage is currently (%69) not so high.  

More users will try EModelRunner now since the Synplas paper is released.
Increasing the coverage will allow us detect issues before them.

Most tests are functional, we need more unit tests.

**Comment by anilbey** [comment]
https://github.com/BlueBrain/EModelRunner/pull/56 references this issue.

**Comment by AurelienJaquier** [comment]
One quick way to increase coverage would be to omit from coverage report the stuff we don't plan on covering anyway. I am thinking of the GUI, or synapses/minis.py.
Aside from that, I agree that we need more unit tests.

**Comment by anilbey** [comment]
@AurelienJaquier, can we instead add unit tests for the Tkinter GUI at least for its main functions like clicking the start button to run the simulation? GUI module depends on many modules like morphologies, recordings, synapses but the changes made in those modules are not testing if the GUI functions are still compatible.

It seems there are event methods to interact with UI elements, e.g. do a click, enter some value
`self.button.event_generate('<Enter>', x=0, y=0)`

This is a test file for Tkinter:
https://github.com/python/cpython/blob/main/Lib/idlelib/idle_test/test_tooltip.py#L106
And this is the source code that is tested in the test above:
https://github.com/python/cpython/blob/main/Lib/idlelib/tooltip.py

source: https://www.reddit.com/r/learnpython/comments/hf5uwu/comment/fvvmhsh/?utm_source=share&utm_medium=web2x&context=3

**Comment by anilbey** [comment]
The GUI module also has simulation functions that need to stay compatible with the rest of the modules. I think it would be good to keep them updated with the future changes.

https://github.com/BlueBrain/EModelRunner/blob/main/emodelrunner/GUI_utils/simulator.py#L177

**Comment by AurelienJaquier** [comment]
Thank you for the links, this will help. I'll add some tests for the GUI module then.

**Comment by anilbey** [comment]
https://github.com/BlueBrain/EModelRunner/pull/59 adds unit tests for GUI/simulator

**Comment by AurelienJaquier** [comment]
I am afraid we won't be able to test the frames, buttons, and other tkinter stuff. For that, we need a graphical environment, and it looks like we don't have one in github actions (source: https://github.com/python/cpython/blob/main/Lib/idlelib/idle_test/README.txt#L47).

**Comment by wvangeit** [comment]
Yeah, not sure if we need to test that. If you really want to, I think @alex4200 has something running on our systems to test GUIs?
